src/Runner.Worker/ActionManager.cs

```diff
@@ -21,11 +21,24 @@
 
 namespace GitHub.Runner.Worker
 {
+    public class PrepareResult
+    {
+        public PrepareResult(List<JobExtensionRunner> containerSetupSteps, Dictionary<Guid, IActionRunner> preStepTracker)
+        {
+            this.ContainerSetupSteps = containerSetupSteps;
+            this.PreStepTracker = preStepTracker;
+        }
+
+        public List<JobExtensionRunner> ContainerSetupSteps { get; set; }
+
+        public Dictionary<Guid, IActionRunner> PreStepTracker { get; set; }
```

Comment:
nit: it looks like different layers uses different terminology (`pre` vs `prepare` vs `init`). Same for `post` vs `cleanup`

Reply:
i will change everything to be init and cleanup

Reply:
sorry, `pre` and `post` to lines up with yaml